.github/workflows/build.yml

```diff
@@ -0,0 +1,40 @@
+name: build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test-and-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up go
+        uses: actions/setup-go@v2-beta
+        with:
+          go-version: '1.13'
+      - name: Set up go env
+        run: |
+          echo "::set-env name=GOPATH::$(go env GOPATH)"
+          echo "::add-path::$(go env GOPATH)/bin"
+        shell: bash
+      - name: Install jq
+        run: |
+          mkdir -p deps/bin
+          curl -s -L -o deps/bin/jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64
+          chmod +x deps/bin/jq
+          echo "::add-path::${PWD}/deps/bin"
+      - name: Test
+        run: make test
+      - name: Build
+        run: |
+          make build-linux
+          make package-linux
+      - uses: actions/upload-artifact@v1
+        with:
+          name: lifecycle
```

Comment:
Should this at least be named `lifecycle-linux-x86-64` to future proof against Windows?

Reply:
Perhaps, I feel like by the time we arrive at that world, this will be versioned anyway and the format will have to change regardless, so did the minimal to get things moving.

Reply:
I agree with @jromero on this one